### ISSUE TITLE
Unexpected version in generated url

### ISSUE THREAD

**Issue by Roms1383** [PaulHatch/semantic-version#59]
Hi, sorry for bothering again !
I actually stumbled a small edge case that I'm aware is probably very unusual.

So here my current release flow : I use this action to detect the next semantic version, chaining it with [JasonEtco/build-and-tag-action@v2](https://github.com/JasonEtco/build-and-tag-action) to produce a packaged release.

And here's the edge case : the way `JasonEtco/build-and-tag-action@v2` works, it will take the release, package the files (using Javascript Github Action in combination with `@vercel/ncc`), move the git tag to the new commit produced, on a separate commit not part of git history. So next time this action runs, it cannot find any previous git tags since the other action moves them out of reach, and of course it ends up telling me it cannot find any previous releases.

My idea was to use `tag_prefix: 'rc'` input to scan previous git tags with your action, while having it generate a link for `vX.X.X`, by using it in combination with `version_format: "v${major}.${minor}.${patch}"`. My guess here is that `version_format` is correct in the outputs, but _it won't modify the generated url_.

Basically I need this because `JasonEtco/build-and-tag-action@v2` will always mess around with git tags, moving them out of reach of `paulhatch/semantic-version@v5.0.0-alpha2` scanning. So as you can guess my idea is to scan `rc` prefixed versions from `main` branch, while letting the other final action do whatever it has to with `v` prefixed versions.

Would you consider adding a mechanism to update the generated url so that, while scanning for `rc` prefixed previous git tags, it would still offer a way to generate a link to create a `v` prefixed release ?

If it's too niche, I can utterly understand though.

**Comment by Roms1383** [comment]
Also realizing that I might not be using both of these actions the way they are meant to. So if ever you get a better alternative that doesn't require any change on your part, I'm utterly interested.

**Comment by PaulHatch** [comment]
Sorry, I'm not familiar with this build-and-tag action, but if I understand what you're saying on this line correctly:

> move the git tag to the new commit produced, on a separate commit not part of git history

That sounds a bit odd and quite "proprietary". Sounds a bit awkward to make these two fit together. I'm not sure I follow 100%, but if you're talking about the URL that the action outputs into the logs, that's supposed to match the "watched" tag. (Note that there are two concepts here, the version tag is not the same as the output version. Semantic versions for example do not have a prefix, per the semver spec, while tags can and often do.

You could output your own link in a simple action using the output from the action, but I'd also recommend looking into the new(ish) manual approval features that have been added to GitHub Actions since this action was created and consider using an "approve deploy" step instead of using the URL. I probably would not have added the URL to the log if this feature had been available at the time the action was created.

**Comment by Roms1383** [comment]
> move the git tag to the new commit produced, on a separate commit not part of git history

As you can see from my repo, the `build-and-tag` action will generate files to publish (`action.yml` and generated `dist/index.js` file). While doing so, it commits **only these files** and move the git tag along there, in a separate commit out of history (as [can be seen for example here](https://github.com/Roms1383/emissary/commit/0e5e3771ef538d0fe4815efd77a98ceb53e84987), on my latest release). This is what they explain in the first lines [on their README.md](https://github.com/JasonEtco/build-and-tag-action).

> That sounds a bit odd and quite "proprietary", sounds a bit awkward to make these two fit together.

As far as I understand, they do so to keep track of the generated files in a reliable way. If you create a traditional release and publish the generated files by yourself, then your action consumers have to trust that these generated files actually comes from your repo files at this specific point in history, and if ever they don't, they would have to manually generate them and compare them by themselves to make sure they match.

> Note that there are two concepts here, the version tag is not the same as the output version. Semantic versions for example do not have a prefix, per the semver spec, while tags can and often do.

On this point I trust you since you definitely know better than I do.

> You could output your own link in a simple action using the output from the action

Indeed that's what I had in mind ultimately. This _two releases mode_ (using `rcx.x.x` on `main` branch to allow your action to properly deduce the next tag, while using `vx.x.x` for releases as `build-and-tag` action moves the git tags around) is not great but necessary since `build-and-tag` action modifies history (even if for good reasons though).

> but I'd also recommend looking into the new(ish) manual approval features that have been added to GitHub Actions.

Ah ? I wasn't aware of this, I definitely need to look into it.

Thanks for your feedback @PaulHatch
